### PR TITLE
[BPK-3118] Support dark mode in BPKButton

### PR DESCRIPTION
**Pull request by georgegillams**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/master/Backpack/Backpack.h)
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

Simulator Screen Shot - iPhone 8 - 2019-10-28 at 11 57 59
Simulator Screen Shot - iPhone 8 - 2019-10-28 at 11 57 57
Simulator Screen Shot - iPhone 8 - 2019-10-28 at 11 57 55
Simulator Screen Shot - iPhone 8 - 2019-10-28 at 11 57 54
Simulator Screen Shot - iPhone 8 - 2019-10-28 at 11 57 52
Simulator Screen Shot - iPhone 8 - 2019-10-28 at 11 57 50

### CODE REVIEW

**Comment by k0nserv** [comment]
Can you explain the use of `forceUserInterfaceStyleLight` and `unforceUserInterfaceStyleLight`?

**Comment by georgegillams** [comment]
> Can you explain the use of `forceUserInterfaceStyleLight` and `unforceUserInterfaceStyleLight`?

Lol no not really

**Comment by georgegillams** [comment]
Actually tbh I don't think we need to handle different button styles differently...

**Comment by imchristian** [comment]
This PR is ready for review? I think that if the background is not going to change depending on the OS mode context then I'm happy with this approach

**Comment by georgegillams** [comment]
Just checking with design

File: Backpack/Button/Classes/BPKButton.m [hunk 1]
```diff
@@ -42,8 +42,13 @@ @interface BPKButton ()
 @property(nonatomic, class, readonly) UIColor *highlightedOutline;
 @property(nonatomic, class, readonly) UIColor *highlightedBlue;
 @property(nonatomic, class, readonly) UIColor *highlightedRed;
+@property(nonatomic, class, readonly) UIColor *disabledLinkColor;
```
Comment (imchristian) [inline Backpack/Button/Classes/BPKButton.m]: Why these colors are `class` properties? are we going to use them outside a particular `BPKButton` instance?

File: Example/SnapshotTests/BPKSnapshotTest.h [hunk 2]
```diff
@@ -77,3 +77,47 @@
 }
 #endif
 
+#if __BPK_DARK_MODE_SUPPORTED
```
Comment (imchristian) [inline Example/SnapshotTests/BPKSnapshotTest.h]: Can you give me a little bit of context to understand what these new methods are doing?

File: UNRELEASED.md [hunk 3]
```diff
@@ -13,8 +13,8 @@
 - Backpack/TextView:
   - Fix to be dark mode compatible
 
-- Backpack/Calendar:
-  - Fix to be dark mode compatible
+- Backpack/Button:
```
Comment (imchristian) [inline UNRELEASED.md]: Bad merge maybe? 🤔

File: package.json [hunk 4]
```diff
@@ -40,7 +40,7 @@
   },
   "homepage": "https://github.com/Skyscanner/backpack-ios#readme",
   "devDependencies": {
-    "bpk-svgs": "^8.2.7",
+    "bpk-svgs": "^8.2.5",
```
Comment (imchristian) [inline package.json]: Bad merge maybe?

**Pull request merged**